XWG8AW/Infrastructure/QuestionDeserializer.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading.Tasks;
+
+using XWG8AW.Domain;
+
+namespace XWG8AW.Infrastructure
+{
+    internal class QuestionDeserializer
+    {
+        public async Task<List<QuestionJson>> QuestionDeserializeFromJson()
+        {
+            int debugFolderLength = 16;
+            string fullpath = Environment.CurrentDirectory;
+            string path = fullpath.Substring(0, fullpath.Length - debugFolderLength);
+            string correctPath = string.Concat(path, "questions.json");
+
+            try {
+                using (var stream = File.OpenRead(correctPath))
+                {
+                    List<QuestionJson>? questions = await JsonSerializer.DeserializeAsync<List<QuestionJson>>(stream, new JsonSerializerOptions
+                    {
+                        AllowTrailingCommas = true,
+                    });
+                    if (questions is null)
+                    {
+                        Console.WriteLine("Deszerializacios hiba");
+                        return null;
+                    }
+                    /*foreach (var question in questions)
+                    {
+                        Console.WriteLine(question);
+                    }*/
+
+                    return questions;
+                }
+
+            } catch (Exception ex) { 
```

Comment:
when keyword-el lehet exception filtert is írni.

Reply:
Köszönöm a tippet. Beismerem ez a megvalósítás az első Stack Overflow találat alapján jött létre.